lib/glific/messages.ex

```diff
@@ -190,8 +191,20 @@ defmodule Glific.Messages do
   end
 
   @doc false
-  @spec create_and_send_message(map()) :: {:ok, Message.t()}
+  @spec create_and_send_message(map()) :: {:ok, Message.t()} | {:error, String.t()}
   def create_and_send_message(attrs) do
+    contact = Glific.Contacts.get_contact!(attrs.receiver_id)
+
+    case attrs[:is_hsm] do
+      true -> Contacts.can_send_hsm_message_to?(contact)
+      _ -> Contacts.can_send_message_to?(contact)
+    end
```

Comment:
This is probably better off as:

can_send_message_to?(contact, is_hsm?)

and use. a when clause there, avoids case and/or if statement in multiple places

Reply:
Right, will do that.

Reply:
@dlobo made the changes. Can you please have a look now?